packages/remix/src/utils/web-fetch.ts

```diff
@@ -70,12 +70,12 @@ export const getSearch = (parsedURL: URL): string => {
 export const normalizeRemixRequest = (request: RemixRequest): Record<string, any> => {
   const { requestInternalsSymbol, bodyInternalsSymbol } = getInternalSymbols(request);
 
-  if (!requestInternalsSymbol) {
-    throw new Error('Could not find request internals symbol');
+  if (!requestInternalsSymbol && !request.headers) {
+    throw new Error('Could not find request headers');
   }
 
-  const { parsedURL } = request[requestInternalsSymbol];
-  const headers = new Headers(request[requestInternalsSymbol].headers);
+  const parsedURL = requestInternalsSymbol ? request[requestInternalsSymbol].parsedURL : new URL(request.url);
+  const headers = requestInternalsSymbol ? new Headers(request[requestInternalsSymbol].headers) : request.headers;
```

Comment:
The standard fetch API should always work (with polyfill or not), so I think the code can be cleaner if `getInternalSymbols` and the associated code logic is deleted.

Thoughts?

Reply:
We had issues with extracting `headers` in previous Remix versions, and we need to stay backwards compatible. For reference: https://github.com/getsentry/sentry-javascript/issues/6139#issuecomment-1315799722

Reply:
@onurtemizkan thanks for the reference. Yes - Remix uses a Proxy but as long as the code sticks to the Standard API interface, it shouldn't cause problems.

I understand that the issues happened before this layer was added, per your commit at https://github.com/getsentry/sentry-javascript/commit/0ee35f070927b47a37270d6037a3f004f3bce492. I do not know the whole code, but from what I see in this layer, `request.headers` is only used in this function, and the usage seems consistent with the standard API. I wouldn't expect it to introduce any problems if the code is changed to use the standard API in this `normalizeRemixRequest` function. Is this wrong?

Reply:
The problem is that we had been reading `req.headers` directly before https://github.com/getsentry/sentry-javascript/commit/0ee35f070927b47a37270d6037a3f004f3bce492. We still provide official support to the `remix` version and `web-fetch` version (as it's dependency) that caused #6139.

We don't rely on those versions being aligned to the current Standard API Interface (which was the reason we ended up vendoring code from them). So, IMO it's not worth risking to revive old resolved issues if this patch resolves the non-polyfilled usage. If we decide in the future to drop support for Remix v1 or Node versions < 18 in the Remix SDK, we can work on a broader cleanup, deleting vendored code (which we generally use as the last resort).

Reply:
> The problem is that we had been reading req.headers directly before https://github.com/getsentry/sentry-javascript/commit/0ee35f070927b47a37270d6037a3f004f3bce492.

Yes - what I mean is that while I do not know the code usage before https://github.com/getsentry/sentry-javascript/commit/0ee35f070927b47a37270d6037a3f004f3bce492, I can see how the `normalizeRemixRequest` function added in https://github.com/getsentry/sentry-javascript/commit/0ee35f070927b47a37270d6037a3f004f3bce492 commit is reading the headers. I do not think it should create any problems or break backward compatibility.

> So, IMO it's not worth risking to revive old resolved issues if this patch resolves the non-polyfilled usage.
Hopefully, this is guaranteed by a test 😅.

All good to leave that cleanup for later!